presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -73,12 +81,45 @@ public String explain(CreateTable statement, List<Expression> parameters)
         }
 
         List<ColumnMetadata> columns = new ArrayList<>();
+        Map<String, Object> finalProperties = new HashMap<>();
+        boolean foundIncluding = false;
         for (TableElement element : statement.getElements()) {
-            Type type = metadata.getType(parseTypeSignature(element.getType()));
-            if ((type == null) || type.equals(UNKNOWN)) {
-                throw new SemanticException(TYPE_MISMATCH, element, "Unknown type for column '%s' ", element.getName());
+            if (element instanceof ColumnDefinition) {
+                ColumnDefinition column = (ColumnDefinition) element;
+                Type type = metadata.getType(parseTypeSignature(column.getType()));
+                if ((type == null) || type.equals(UNKNOWN)) {
+                    throw new SemanticException(TYPE_MISMATCH, column, "Unknown type for column '%s' ", column.getName());
+                }
+                columns.add(new ColumnMetadata(column.getName(), type));
+            }
+            else if (element instanceof LikeClause) {
+                LikeClause likeClause = (LikeClause) element;
+                QualifiedObjectName likeTableName = createQualifiedObjectName(session, statement, likeClause.getTableName());
```

Comment:
This needs to verify the LIKE table is in the same catalog as table being created. See `RenameTableTask`